test/models/watching_transaction_test.rb

```diff
@@ -0,0 +1,139 @@
+                    }.to_json
+                  )
+
+    WatchingTransaction.resend_transactions
+
+    assert_requested(stub_post, times: 5)
+    assert_equal 5, WatchingTransaction.where(group_id: original.group_id).count,
+                 'should work'
+  end
+
+  test 'resend transactions should destroy all if one is mined' do
+    original = FactoryBot.create(:watching_transaction)
+    3.times do
+      FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+    end
+    latest = FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+
+    stub_post = stub_request(:post, EthereumApi::SERVER_URL)
+                  .with(body: /eth_getTransactionByHash/)
+                  .to_return(
+                    body: {
+                      result: {
+                        'block_number' => nil,
+                      }
+                    }.to_json
+                  )
+
+    stub_post_one = stub_request(:post, EthereumApi::SERVER_URL)
+                      .with(body: /eth_getTransactionByHash.*#{latest.txhash}/)
+                      .to_return(
+                        body: {
+                          result: {
+                            'block_number' => SecureRandom.hex(8),
+                          }
+                        }.to_json
+                      )
+
+    WatchingTransaction.resend_transactions
+
+    assert_requested(stub_post, times: 5)
+    assert_requested(stub_post_one, times: 1)
+    assert_equal 0, WatchingTransaction.where(group_id: original.group_id).count,
+                 'should destroy all if one is mined'
+  end
+
+  test 'resend transactions should resend latest if rest are dropped' do
+    original = FactoryBot.create(:watching_transaction)
+    2.times do
+      FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+    end
+    sleep 1 # UUID is not sortable, so some delay is needed in order to properly sort via created_at
+    latest = FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+
+    stub_post = stub_request(:post, EthereumApi::SERVER_URL)
+      .with(body: /eth_getTransactionByHash/)
+      .to_return(
+        body: {
+          result: nil
+        }.to_json
+      )
+
+    new_txhash = generate(:txhash)
+    stub_post_send = stub_request(:post, EthereumApi::SERVER_URL)
+      .with(body: /eth_sendRawTransaction/)
+      .to_return(
+        body: {
+          result: new_txhash
+        }.to_json
+      )
+
+    WatchingTransaction.resend_transactions
+
+    assert_requested(stub_post, times: 4)
+    assert_requested(stub_post_send, times: 1)
+    assert_equal 1, WatchingTransaction.where(group_id: original.group_id).count,
+                 'should work'
+
+    resent = WatchingTransaction.find_by(id: latest.id)
+    assert_equal new_txhash, resent.txhash,
+                 'should update txhash for resent transaction'
+  end
+end
```

Comment:
Two tests left: 
1. To handle if `get_transaction` returns an HTTP error  and method should not raise or throw an error

2. To handle if `send_transaction` returns an HTTP error and should not also raise an error

Reply:
What method should not raise or throw an error?
request_ethereum_server has rescue

Reply:
It is `WatchingTransaction.resend_transaction`. So the concern is if in the scheduler there is an error if the API is down, it should be covered and not cause an error. Those two new tests should cover it.

In terms of assertion, you can just assert the errored stub should be called.

Reply:
@FrancisMurilloDigix should we break resend_transactions in case of API error? Or just go to the next iteration?

Reply:
And I'm really not getting why these tests are necessary.
What you are talking about is request_ethereum_server responsibilities.
The only thing I see is necessary here is handling :ok vs :error result, because right now if result is :error then transaction is considered dropped, which is wrong behaviour if API is just unavailable. That's why I'm asking whether we, in this case, should break job at all and wait 5 minutes for the next one or we should break only current iteration where API error has occurred?

Reply:
And right now if send_raw_transaction has failed, then it'll just wait for a new attempt in 5 minutes